### ISSUE TITLE
Endar Spire - Various issues

### ISSUE THREAD

**Issue by DarthParametric** [KOTORCommunityPatches/K1_Community_Patch#490]
1.9.0 beta test observations:

Command Module:
- [ ] ~The intro cutscene (post-movie) seems to have the music cut out and then start again from the beginning. Have not experienced that previously.~ Edit: This might be a specific hardware/software combo issue, not the game.
- [x] Trask's head model has some clipping issues with the right eyelid in its open state.
- [x] Trask's head model has some shading issues around the outer corner of the right eye.
- [x] Multiple synced death cries from the Republic soldiers. Swap a few of them to different soundsets, stagger death timings.
- [x] The level-up DLG that pops up on via the OnUsed of the exit door of the bridge has a long delay before it initiates. The camera angle is also too close, try angle 2 or 3. Could probably use a party jump.
- [ ] ~After killing the two mooks in the doorway of the bridge, there's a forced conversation with Trask that fires ("Bastila's not here") while there are scripted mooks fighting further in. They die off-screen while Trask is yapping. Their death should be the first shot before Trask's lines kick in.~ Edit: Tried it, but for brevity it's best to leave them getting nuked in the background.
- [x] The player standing in the bridge doorway may prevent it from closing as intended (per #563).

Starboard Section:
- [x] When meeting Carth in the final room, he needs a facing towards the player for his triggered DLG.

**Comment by DarthParametric** [comment]
1.10.0 beta test observations:

- [ ] The new death cries are a mixed bag. Some didn't seem to fire, some are pretty low key. Can probably add back one or two of the vanilla Rep trooper soundset to the second cutscene.
- [ ] There are Sith trooper rotting corpse placeables used in both modules which look out of place.

**Comment by DarthParametric** [comment]
1.10.0 beta test 2 observations:

- [ ] Trask needs an animation when he is made to open the door in the first room.
- [ ] The doors in the Starboard section have low health.

**Comment by Salk73** [comment]
I have an additional observation.

We should change name for rsldcrps002 to 25898.

**Comment by DarthParametric** [comment]
That [external script call](https://github.com/KOTORCommunityPatches/Vanilla_KOTOR_Script_Source/blob/master/K1/Data/scripts.bif/k_pman_npcstart.nss) just turns Trask's AI back on.

As to him failing to move, was he party leader? Although I assume the scene should have switched the player back to leader at the start. Might be better to give him a `ExecuteScript("k_ai_master", oTrask, 3001)` i.e. a `GN_DetermineCombatRound` rather than using `ActionAttack`. Especially since he is presumably still armed with a blaster.

**Comment by Salk73** [comment]
No, he was not party leader. The door was opened by the protagonist.

Trask could be stripped of his weapons by the player in theory, but he should be wielding his blaster at that point.

The cutscene is handled awkwardly from the start. The protagonist is jumped to the corridor where the battle is taking place and Trask talks standing on the other side of the door.

**Comment by Salk73** [comment]
If the party leader after killing the two sith soldiers in the Endar Spire bridge is Trask then we have a facing problem in the following cutscene where I believe Trask is talking to the invisible placeable. I remedied by adding a `SetPartyLeader(-1);` in the k_pend_room7_01 OnDeath above the `SignalEvent(oInvis, EventUserDefined(100));` line.
